### PR TITLE
[ottf] add non-volatile scratch region

### PR DESCRIPTION
**Pull request by timothytrippel**
This adds a region in flash to write/read non-volatile data to
persist accross chip resets.

Signed-off-by: Timothy Trippel <ttrippel@google.com>

### CODE REVIEW

File: sw/device/tests/example_test_from_flash.c [hunk 1]
```diff
@@ -70,6 +70,14 @@ const test_config_t kTestConfig = {
 // void ottf_timer_isr(void) {}
 // void ottf_external_isr(void) {}
 
+/**
+ * Place data in flash that will need to persist across resets by marking with
+ * with the section ".non_volatile_scratch"). Write to this region with the
+ * flash controller DIF.
+ */
```
Comment (tjaychen) [inline sw/device/tests/example_test_from_flash.c:77]: should we also say read to this section with `mmio`?

Comment (timothytrippel) [reply sw/device/tests/example_test_from_flash.c:77]: Done.

**Review by tjaychen — APPROVED** [review]
thanks for adding this! much appreciated!

File: sw/device/tests/example_test_from_flash.c (hunk 1 above)
Comment (a-will) [reply sw/device/tests/example_test_from_flash.c:77]: You'd actually use `abs_mmio_read32(...)` or similar. :)

Comment (timothytrippel) [reply sw/device/tests/example_test_from_flash.c:77]: Fixed.

**Pull request merged**